sql/core/src/main/scala/org/apache/spark/sql/api/python/ChunkReadUtils.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.api.python
+
+import java.io.ByteArrayOutputStream
+
+import scala.collection.mutable.ArrayBuffer
+import scala.jdk.CollectionConverters._
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.execution.arrow.ArrowBatchStreamWriter
+import org.apache.spark.sql.types.{DataType, StructType}
+import org.apache.spark.storage.{ArrowBatchBlockId, BlockId, StorageLevel}
+
+
+case class ChunkMeta(
+  id: String,
+  rowCount: Long,
+  byteCount: Long
+)
+
+object ChunkReadUtils {
+
+  def persistDataFrameAsArrowBatchChunks(
+      dataFrame: DataFrame, maxRecordsPerBatch: Int
+  ): Array[ChunkMeta] = {
+    val sparkSession = SparkSession.getActiveSession.get
+    val rdd = dataFrame.toArrowBatchRddWithBatchRowCount(maxRecordsPerBatch)
```

Comment:
We only call `toArrowBatchRddWithBatchRowCount` once. Shall we inline it? We split arrow batches and save them to the block manager at the same time.